alternator/rjson.cc

```diff
@@ -113,6 +113,48 @@ void push_back(rjson::value& base_array, rjson::value&& item) {
 
 }
 
+bool single_value_comp::operator()(const rjson::value& r1, const rjson::value& r2) const {
+       auto r1_type = r1.GetType();
+       auto r2_type = r2.GetType();
+       switch (r1_type) {
+       case rjson::type::kNullType:
+           return r1_type < r2_type;
+       case rjson::type::kFalseType:
+           return r1_type < r2_type;
+       case rjson::type::kTrueType:
+           return r1_type < r2_type;
+       case rjson::type::kObjectType:
+           throw rjson::error("Object type comparison is not supported");
+       case rjson::type::kArrayType:
+           throw rjson::error("Array type comparison is not supported");
+       case rjson::type::kStringType: {
+           const size_t r1_len = r1.GetStringLength();
+           const size_t r2_len = r2.GetStringLength();
+           size_t len = std::min(r1_len, r2_len);
+           int result = std::strncmp(r1.GetString(), r2.GetString(), len);
+           return result < 0 || (result == 0 && r1_len < r2_len);
+       }
+       case rjson::type::kNumberType: {
+           if (r1_type != r2_type) {
+               throw rjson::error("All numbers in a set should have the same type");
```

Comment:
What "set"?
And why is this r1_type != r2_type done only for numbertype, and not as the first thing in this function?
And again, you'll need to catch this exception so the user don't see internal errors).

Reply:
>  And why is this r1_type != r2_type done only for numbertype, and not as the first thing in this function?

That's not that simple, since some types can compare against each other, e.g. `kFalseType < kTrueType`. It should be present for most cases though.

Reply:
and mentioning "set" here is obviously a mistake as well - I guess I started with coding this struct as a local way of providing a comparison operator for a set, but then decided to make it more general.